src/libcore/convert.rs

```diff
@@ -45,6 +69,12 @@ use marker::Sized;
 /// let s = "hello".to_string();
 /// is_hello(s);
 /// ```
+///
+/// # Generic Impls
+///
+/// - `AsRef` auto-dereference if the inner type is a reference or a mutable
+/// reference
```

Comment:
Yep, I think so. It's part of the blanket impls. [ref](https://github.com/rust-lang/rust/pull/30901/files#diff-da707dc4d90c85e7074dbdecebfe8785R178)